packages/x-date-pickers-pro/src/internal/hooks/useDesktopRangePicker/useDesktopRangePicker.tsx

```diff
@@ -96,22 +100,12 @@ export const useDesktopRangePicker = <
     });
   };
 
-  const fieldSlotProps = useRangePickerInputProps({
-    wrapperVariant: 'desktop',
-    open,
-    actions,
-    readOnly,
-    disabled,
-    disableOpenPicker,
-    localeText,
-    onBlur: handleBlur,
-    rangePosition,
-    onRangePositionChange,
-  });
-
   const Field = slots.field;
-  const fieldProps: BaseMultiInputFieldProps<
+  const fieldType = (Field as any).isSingleInput ? 'single-input' : 'multi-input';
```

Comment:
> We could possibly have this field as the one known Field slot prop

Not sure to understand
I tried to add the `isSingleInput` to the typing of the slot in `RangePickerFieldSlotsComponent` but it causes bugs when you actually try to pass a field to the component. I can dive a little deeper.

> And we should probably opt for more safe approach of using ?. in order not to error out here.

Field can never be undefined here, it I understand which `?` you want to add.

Reply:
> Not sure to understand
> I tried to add the `isSingleInput` to the typing of the slot in `RangePickerFieldSlotsComponent` but it causes bugs when you actually try to pass a field to the component. I can dive a little deeper.

Yes, I meant exactly that, but of course, it could definitely raise other issues somewhere else. So, maybe it's not that bad of an approach. 🤔 
I'm just thinking that maybe we could instead go with an enum for this field, especially when it is somewhat of an internal prop. If it would ever become a case that we'd create yet another input type—this wouldn't be a breaking change. 🤔

Reply:
> I'm just thinking that maybe we could instead go with an enum for this field

Something lilke `SingleInputDateRangeField.fieldType = 'single-input'` ?

Reply:
Yup. Not strictly a requirement, as the likelihood of us introducing yet another input type, is possibly not that viable, but it would possibly not hurt that much having this option open. 🤔